exp/api/v1alpha4/azuremanagedcontrolplane_webhook.go

```diff
@@ -306,3 +307,36 @@ func (r *AzureManagedControlPlane) validateSSHKey() error {
 
 	return nil
 }
+
+// ValidateLoadBalancerProfile validates a LoadBalancerProfile.
+func (r *AzureManagedControlPlane) validateLoadBalancerProfile() error {
+	if r.Spec.LoadBalancerProfile != nil {
+		var allErrs field.ErrorList
+
+		if r.Spec.LoadBalancerProfile.ManagedOutboundIPs != nil {
+			if *r.Spec.LoadBalancerProfile.ManagedOutboundIPs < 1 || *r.Spec.LoadBalancerProfile.ManagedOutboundIPs > 100 {
+				allErrs = append(allErrs, field.Invalid(field.NewPath("Spec", "LoadBalancerProfile", "ManagedOutboundIPs"), *r.Spec.LoadBalancerProfile.ManagedOutboundIPs, "value should be in between 1 and 100"))
+			}
+		}
+
+		if r.Spec.LoadBalancerProfile.AllocatedOutboundPorts != nil {
+			if *r.Spec.LoadBalancerProfile.AllocatedOutboundPorts < 0 || *r.Spec.LoadBalancerProfile.AllocatedOutboundPorts > 64000 {
```

Comment:
should we even allow 0?

Reply:
According to the [api doc](https://docs.microsoft.com/en-us/rest/api/aks/managed-clusters/create-or-update#managedclusterloadbalancerprofile), the allowed values are in the range of 0 to 64000 (inclusive). Let me know if you think we shouldn't allow users to explicit specify value 0.

Reply:
hmm, I think 0 may indicate to use the static allocation strategy. I guess that's fine 👍